chain/chain/src/runtime/mod.rs

```diff
@@ -710,7 +710,7 @@ impl RuntimeAdapter for NightshadeRuntime {
                 storage_config.use_flat_storage,
             ),
         };
-        if checked_feature!("stable", StatelessValidationV0, PROTOCOL_VERSION)
+        if checked_feature!("stable", StateWitnessSizeLimit, protocol_version)
```

Comment:
I think that we need to enable recording when either this ***or next*** epoch has protocol version with stateless validation enabled. We need to save `StateTransitionData` from the current epoch to produce a `ChunkStateWitness` in the next one.

Reply:
Gotcha, will do and add a comment! 

It would be a total overkill but we could consider using the multi-protocol-upgrades in single release to do this. (Once it's implemented).

Reply:
By the way, do we garbage collect `StateTransitionData`?

Reply:
> By the way, do we garbage collect `StateTransitionData`?

We do, Anton added it in https://github.com/near/nearcore/pull/10599
And AFAIK it happens even on archival nodes x.x